### PR TITLE
Generate sample cluster.yml

### PR DESCRIPTION
**Pull request by kamidzi**
A small convenience for printing a sample cluster.yml rather than having to copy the mac addresses. To build on ubuntu, it's important to install the following packages before atttempting to install `virtualbox`:

`pcregrep make gcc`

This was tested with following versions of bash:
```
GNU bash, version 4.3.11(1)-release (x86_64-pc-linux-gnu)
GNU bash, version 4.3.42(1)-release (x86_64-apple-darwin14.5.0)
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin15)
```

### CODE REVIEW

File: bootstrap/ansible_scripts/scripts/spawn_local_vms.sh [hunk 1]
```diff
@@ -154,8 +156,33 @@ done
...
+                                  [bcpc-vm2]='10.0.100.12'
+                                  [bcpc-vm3]='10.0.100.13')
 for VM in bcpc-bootstrap $VMS; do
   MAC_ADDRESS=$(VBoxManage showvminfo --machinereadable $VM | pcregrep -o1 -M '^hostonlyadapter\d="vboxnet0"$\n*^macaddress\d="(.+)"' | $SED 's/^(..)(..)(..)(..)(..)(..)$/\1:\2:\3:\4:\5:\6/')
-  echo "MAC address for $VM is $MAC_ADDRESS"
+  cat << EoF
+  $VM:
+    domain: bcpc.example.com
+    hardware_type: Virtual
+    ip_address: ${DEFAULT_ROLE_MAPPING[$VM]}
```
Comment (erhudy) [inline bootstrap/ansible_scripts/scripts/spawn_local_vms.sh]: I think this should be `ip_address: ${DEFAULT_IP_MAPPING}`. There also appears to be an issue where it's designating every vm as the **work-ephemeral** type. This is with bash 3.2.57.
```
---
cluster_name: Test-Laptop-Ansible
nodes:
  bcpc-bootstrap:
    domain: bcpc.example.com
    hardware_type: Virtual
    ip_address: work-ephemeral
    ipmi_address:
    mac_address: 08:00:27:DC:F9:91
    role: work-ephemeral
  bcpc-vm1:
    domain: bcpc.example.com
    hardware_type: Virtual
    ip_address: work-ephemeral
    ipmi_address:
    mac_address: 08:00:27:79:58:83
    role: work-ephemeral
  bcpc-vm2:
    domain: bcpc.example.com
    hardware_type: Virtual
    ip_address: work-ephemeral
    ipmi_address:
    mac_address: 08:00:27:36:C3:94
    role: work-ephemeral
  bcpc-vm3:
    domain: bcpc.example.com
    hardware_type: Virtual
    ip_address: work-ephemeral
    ipmi_address:
    mac_address: 08:00:27:32:87:84
    role: work-ephemeral
```

**Comment by erhudy** [comment]
Commented on an issue inline. After making the one fix to switch to `DEFAULT_IP_MAPPING`, it gives all VMs the last IP address, so it seems like bash is using the last value in each hash for everything.

**Comment by erhudy** [comment]
Tested with bash 4 as well and got the same result.

**Comment by erhudy** [comment]
Pushed up a changeset that replaces the associative arrays with functions. It seems to work as intended with bash 3 and 4:
```
---
cluster_name: Test-Laptop-Ansible
nodes:
  bcpc-bootstrap:
    domain: bcpc.example.com
    hardware_type: Virtual
    ip_address: 10.0.100.3
    ipmi_address:
    mac_address: 08:00:27:04:D7:52
    role: bootstrap
  bcpc-vm1:
    domain: bcpc.example.com
    hardware_type: Virtual
    ip_address: 10.0.100.11
    ipmi_address:
    mac_address: 08:00:27:4E:66:09
    role: head
  bcpc-vm2:
    domain: bcpc.example.com
    hardware_type: Virtual
    ip_address: 10.0.100.12
    ipmi_address:
    mac_address: 08:00:27:C3:EC:8F
    role: work
  bcpc-vm3:
    domain: bcpc.example.com
    hardware_type: Virtual
    ip_address: 10.0.100.13
    ipmi_address:
    mac_address: 08:00:27:C6:2B:ED
    role: work-ephemeral
```

**Comment by kamidzi** [comment]
Hey. No kidding... that's probably why I put that comment about bash 4isms... as I filed this I was wondering 'why?'. Now we know. Initially, I was trying to avoid the function approach, but since it seems that bash3 is the default on Mac OS X -> no real debate here.

**Pull request merged**